ip-utils.c

```diff
@@ -12,13 +12,45 @@
  */
 
 #include <errno.h>
+#include <sys/socket.h>
 #include <sys/types.h>
 #include <arpa/inet.h>
+#include <netdb.h>
 #include <net/if.h>
 #include <string.h>
+#include <unistd.h>
+
+#include <glib.h>
 
 #include "ip-utils.h"
 
+int janus_network_detect_local_ip(char *buf, size_t buflen) {
+	struct sockaddr_in addr;
+	socklen_t len;
+	int fd = socket(AF_INET, SOCK_DGRAM, 0);
+	if(fd == -1)
+		goto error;
+	addr.sin_family = AF_INET;
+	addr.sin_port = htons(1);
+	inet_pton(AF_INET, "1.2.3.4", &addr.sin_addr.s_addr);
+	if(connect(fd, (const struct sockaddr*) &addr, sizeof(addr)) < 0)
```

Comment:
What exactly are you trying to do? If all you want to know is the local IP address you could simply walk a `struct * ifaddrs`, filtering for the correct `AF_*` type and maybe checking the address is not link-local (in case of IPv6)?

Also: this function lacks sanity checks for multihomed setups (multiple network interfaces) if you are interested in being able to restrict traffic from Janus to peers to specific network interfaces